src/js/main.js

```diff
@@ -41,22 +45,94 @@ Site.Product = {
     var _this = this;
 
     if ($('#product-gallery').length) {
-      _this.setGalleryWidth();
+      $('#product-gallery').imagesLoaded( function() {
+        _this.setGalleryDimensions();
+      });
+    }
+
+    if ($('#product-select').length) {
+      _this.productSelect();
     }
 
     if ($('#related-products').length) {
       _this.pickRelated();
     }
   },
 
-  setGalleryWidth: function() {
+  setGalleryDimensions: function() {
+    var _this = this;
     var galleryWidth = 0;
+    var windowWidth = $(window).width();
 
-    $('.product-gallery-item').each(function() {
+    $.each($('.product-gallery-item'), function() {
```

Comment:
why did you switch the each here? The selector is jquery so can just each itself